core/src/tpu.rs

```diff
@@ -111,6 +111,7 @@ pub struct Tpu {
 }
 
 impl Tpu {
+    #[deprecated(since = "2.3.0", note = "Use new_with_client instead.")]
```

Comment:
@KirillLykov why deprecate this instead of just deleting it? This is `core`, we don't need to provide a stable interface.

Reply:
i thought for all the pub code we should, i will remove it in this case

Reply:
https://github.com/anza-xyz/agave/pull/6223